php/class-media.php

```diff
@@ -244,6 +244,26 @@ public function is_media( $attachment_id ) {
 		return $is_media;
 	}
 
+	/**
+	 * Check if the attachment is local.
+	 *
+	 * @param int $attachment_id The attachment ID to check.
+	 *
+	 * @return bool
+	 */
+	public function is_local_media( $attachment_id ) {
+		$is_local_media = false;
+
+		$local_host = wp_parse_url( get_site_url(), PHP_URL_HOST );
+		$media_host = wp_parse_url( get_the_guid( $attachment_id ), PHP_URL_HOST );
+
+		if ( $local_host === $media_host ) {
+			$is_local_media = true;
+		}
+
+		return $is_local_media;
```

Comment:
@pereirinha Wouldn't this be a little simpler?
```suggestion
		return $local_host === $media_host;
```